bigchaindb/web/views/transactions.py

```diff
@@ -39,13 +46,32 @@ def get(self, tx_id):
         Return:
             A JSON string containing the data about the transaction.
         """
+        parser = reqparse.RequestParser(trim=True)
+        parser.add_argument(
+            'status',
+            default='valid',
+            type=str,
+            choices=TRANSACTION_STATUSES,
+            case_sensitive=False,
+            help='Transaction status.',
+        )
+        # TODO error handling
+        args = parser.parse_args(strict=True)
+        requested_status = args['status']
+
         pool = current_app.config['bigchain_pool']
 
         with pool() as bigchain:
-            tx = bigchain.get_transaction(tx_id)
+            tx, status = bigchain.get_transaction(tx_id, include_status=True)
 
-        if not tx:
-            return make_error(404)
+        if not tx or status != requested_status:
+            return make_error(
+                404,
+                message=(
+                    'Transaction with id: "{}" and '
+                    'status: "{}" could not be found.'
+                ).format(tx_id, requested_status)
+            )
```

Comment:
I guess this could be mildly surprising for someone who didn't explicitly request a `status` to be notified that there aren't any transactions with `status: valid`, but I think that's fine assuming our HTTP documentation says it defaults to `valid` statuses.